File: PixelDungeonPhotoMaker/src/au/net/genesis/mds/particles/PopUp.java
package au.net.genesis.mds.particles;

public class PopUp extends ParticleLogic{

	
	@Override
	public void initialise() {
		super.initialise();
		alpha = 0;
		xscale = yscale = 0.4F;
		lifeTime = 30;
	}
	
	@Override
	public void update() {
		super.update();
		if (age < lifeTime/2) {
			alpha = Math.min(1F , alpha + 0.1F);
			yscale = xscale += 0.15;
		} else {
			alpha = Math.max(0F , alpha - 0.1F);
			yscale = xscale = Math.max(xscale - 0.15F, 1F);
		}
	}
	
}

File: src/main/java/com/mdsimmo/particles/PopUp.kt
package com.mdsimmo.particles

class PopUp : ParticleLogic() {

    override fun newParticle(): ParticleState {
        return ParticleState(
            alpha = 0f,
            yscale = 0.4f,
            xscale = 0.4f,
            lifeTime = 30
        )
    }

    override fun update(particle: ParticleState): ParticleState {
        val p = super.update(particle)
        return if (p.age < p.lifeTime / 2) {
            p.copy(
                    alpha = Math.min(1f, p.alpha + 0.1f),
                    xscale = p.xscale + 0.15f,
                    yscale = p.yscale + 0.15f
            )
        } else {
            p.copy(
                    alpha = Math.max(0f, p.alpha - 0.1f),
                    xscale = p.xscale - 0.15f,
                    yscale = p.yscale - 0.15f
            )
        }
    }
}